usr.sbin/inetd/inetd.c

```diff
@@ -2674,7 +2674,8 @@ rl_process(struct servtab *sep, int ctrl)
 	    "%zu and se_count %zu", SERV_PARAMS(sep),
 	    sep->se_service_max, sep->se_count);
 
-	if (sep->se_count++ == 0) {
+	/* se_count is incremented when rl_process returns 0 */
+	if (sep->se_count == 0) {
```

Comment:
Only increment se_count when successfully passing rate limiting checks, otherwise this is just "packet" rate limiting and won't work very well next to ip rate limiting.